### ISSUE TITLE
If we decrypt a message, we should write the inventory contents to disk

### ISSUE THREAD

**Issue by grant-olson** [Bitmessage/PyBitmessage#465]
My computer died, and I got some duplicate messages in my inbox.  I understand why we only periodically write the full inventory to disk, but I think if we successfully decrypt a message, we should write the inventory entry for that item to disk so we don't duplicate in the inbox, or even worse, rebroadcast a mailing list post or otherwise reprocess the message via an external app.

**Comment by PeterSurda** [comment]
@Atheros1 has this been done?

**Comment by PeterSurda** [comment]
I will verify this myself.

**Comment by PeterSurda** [comment]
@grant-olson , I reviewed the code. I think what happened was that objectProcessor processed the message from the inventory twice. The current code will detect this as a duplicate anyway so I don't think you'll have this problem with a recent version. I did some tests a while ago on an unrelated issue, and when you empty your inventory and then PyBitmessage has to refetch it, it won't generate duplicates in you inbox. 0.4.4 already has such detection. It had a bug (it produced false positives), but @Atheros1 fixed it later in the master branch. So I'm not going to try to reproduce this particular issue.
